crypto/evp/evp_local.h

```diff
@@ -51,8 +52,11 @@ struct evp_cipher_ctx_st {
     int block_mask;
     unsigned char final[EVP_MAX_BLOCK_LENGTH]; /* possible final block */
 
-    /* Provider ctx */
-    void *provctx;
+    /*
+     * Opaque data returned from a providers cipher algorithm implementation
+     * OSSL_FUNC_cipher_newctx()
+     */
+    void *data;
```

Comment:
How about `ciphctx`?

Reply:
So then we end up with ciphctx sitting inside evp_cipher_ctx_st which is still kind of confusing..
I think it was provctx to indicate that it was a provider side ctx..
Should we go with something even more longwinded like provciphctx ?

Reply:
If you prefer the longwinded variant, go for it.  But then, how about `ciphprovctx` to match the already existing names in EVP_PKEY_CTX (less renaming for you to do as well :wink:)

Reply:
But then we are kind of back where we started with a provctx - which is not a good place to be.

Reply:
Ah, so you mean that `provciphctx` seems clearer than `ciphprovctx`?  I didn't quite catch on to that...  it doesn't make a difference for me, but if it does for you, then do it!

Reply:
Anyone else have an opinion?

Reply:
The bikeshed shall be painted lilac!
Erm, I mean, `provciphctx` seems more clear to me than `ciphprovctx`.

Reply:
I'd be perfectly fine with `provctx`.  The point of the issue was that the names were inconsistent `data` in some places and `provctx` in others.  So long as they are bought in line, I'm not fussed which way.

I'm not sure including `ciph` or similar is adding anything here.  In fact it reduces readability IMO.

Reply:
Err that was one of the points, The other point was that it is not the provider context. It is an allocated blob of data (some form of ctx) that resides in an algorithm implementation (which sits inside a provider). provctx is too generic a term for use here.

Reply:
`provdata` perhaps?

Reply:
I would rather avoid `data` in any form for these contexts, as it suggests some form of payload in my mind. See `keydata`

Reply:
`blob` then.  Or `provblob`.

Reply:
algctx or provalgctx?

Reply:
The thing this maps to is PROV_AES_CTX for example.

Reply:
If you want the One Generic Name, `algctx` gets my vote